app/javascript/tests/patient/Patient.test.js

```diff
@@ -0,0 +1,186 @@
+    'TRAVEL FROM AFFECTED COUNTRY OR AREA',
+    'WAS IN HEALTH CARE FACILITY WITH KNOWN CASES',
+    'LABORATORY PERSONNEL',
+    'HEALTHCARE PERSONNEL',
+    'CREW ON PASSENGER OR CARGO FLIGHT'
+]
+
+describe('Patient', () => {
+    it('Properly renders all main components', () => {
+        const wrapper = shallow(<Patient details={mockPatient1} groupMembers={[ mockPatient2 ]} goto={goToMock} hideBody={true}
+            jurisdictionPath="USA, State 1, County 2" authenticity_token={authyToken} />);
+        expect(wrapper.find('#jurisdiction-path').text()).toEqual('Assigned Jurisdiction: USA, State 1, County 2');
+        expect(wrapper.find('#assigned-user').text()).toEqual('Assigned User: 21');
+        expect(wrapper.find('#identification').exists()).toBeTruthy();
+        expect(wrapper.find('#contact-information').exists()).toBeTruthy();
+        expect(wrapper.find('#address').exists()).toBeTruthy();
+        expect(wrapper.find('#arrival-information').exists()).toBeTruthy();
+        expect(wrapper.find('#additional-planned-travel').exists()).toBeTruthy();
+        expect(wrapper.find('#exposure-case-information').exists()).toBeTruthy();
+    });
+
+    it('Properly renders identification section', () => {
+        const wrapper = shallow(<Patient details={mockPatient1} groupMembers={[ ]} hideBody={true}
+            jurisdictionPath="USA, State 1, County 2" authenticity_token={authyToken} />);
+        const section = wrapper.find('#identification');
+        expect(section.find(Row).first().text()).toEqual('Identification: Minnie M Mouse');
+        expect(section.find(Button).length).toEqual(0);
+        identificationFields.forEach(function(field, index) {
+            expect(section.find('.font-weight-normal').at(index).text()).toEqual(field+':');
+        });
+    });
+
+    it('Properly renders contact information section', () => {
+        const wrapper = shallow(<Patient details={mockPatient1} groupMembers={[ ]} hideBody={true}
+            jurisdictionPath="USA, State 1, County 2" authenticity_token={authyToken} />);
+        const section = wrapper.find('#contact-information');
+        expect(section.find(Row).first().text()).toEqual('Contact Information');
+        expect(section.find(Button).length).toEqual(0);
+        contactFields.forEach(function(field, index) {
+            expect(section.find('.font-weight-normal').at(index).text()).toEqual(field+':');
+        });
+    });
+
+    it('Properly renders address section', () => {
+        const wrapper = shallow(<Patient details={mockPatient1} groupMembers={[ ]} hideBody={true}
+            jurisdictionPath="USA, State 1, County 2" authenticity_token={authyToken} />);
+        const section = wrapper.find('#address');
+        expect(section.find(Row).first().text()).toEqual('Address');
+        expect(section.find(Button).length).toEqual(0);
+        expect(section.find('.font-weight-light').at(0).text()).toEqual('1 Hartford Drive');
```

Comment:
I dont know the best practice for tests. Is it considered good practice to hard-code strings like this? Versus
` expect(section.find('.font-weight-light').at(0).text()).toEqual(mockPatient1.address)`

Reply:
I totally get what you're saying for simple cases, but most of these fields involve some sort of formatting or string concatenation (like for dates).  For example, this isn't just a single address field but rather is looking at this span:

`<span className="font-weight-light">
      {this.props.details.address_line_1 && `${this.props.details.address_line_1}`}
      {this.props.details.address_line_2 && ` ${this.props.details.address_line_2}`}
      {this.props.details.foreign_address_line_1 && `${this.props.details.foreign_address_line_1}`}
       {this.props.details.foreign_address_line_2 && ` ${this.props.details.foreign_address_line_2}`}
 </span>`

So we'd either need to add a formatter or logic to test these fields using the mocks or hard code it in.  In previous cases, I have just hard coded but tbh I don't know what the best practice is here

Reply:
Agreed that hardcoding is not ideal because now if the mockPatient changes for any innocent reason, these tests may fail. I'd say you could add a test helper formatter and that would be best.

Reply:
done - added a utils file that contains basic formatters needed